src/index.js

```diff
@@ -0,0 +1,92 @@
+  import React from 'react';
+  import ReactDOM from 'react-dom';
+
+  //start stitch import
+  import { Stitch, GoogleRedirectCredential } from "mongodb-stitch-browser-sdk";
+  //end stitch import
+  
+  //display google auth user information
+  class User extends React.Component {
+
+    constructor(props) {
+      super(props);
+    }
+
+    render() {
+      if (this.props.value === undefined) return null;
+      
+      return <div>
+              <pre>{this.props.value.data.name}</pre>
+              </div>
+    }
+  }
+
+  class Demo extends React.Component {
+
+    constructor(props) {
+      super(props);
+      this.state = { 
+        user: [],
+      }
+    }
+
+    componentDidMount() {
+      this.setupStitch();
+    }
+    
+    //start stitch setup
+    setupStitch() {
+      //copy the name of your google-auth enabled stitch application here
+      //the name of the app will typically be the stitch application name
+      //with a "-"" + random string appended
+      const appName = 'authentication_test-htbrq';
+
+      //need a description of how this works as the API method names are a
+      //little unclear
+      const stitchClient = Stitch.hasAppClient(appName) ? Stitch.defaultAppClient : Stitch.initializeDefaultAppClient(appName);
+      
+      //check if this user has already authenticated and we're
+      //here from the redirect. If so, process the redirect.
+      if (stitchClient.auth.hasRedirectResult()) {
+          stitchClient.auth.handleRedirectResult().then(user => {
+          console.log("processed redirect result");
+      })}
+
+      //manage user authentication state
+      if (!stitchClient.auth.isLoggedIn) {
+        const credential = new GoogleRedirectCredential();
+        Stitch.defaultAppClient.auth.loginWithRedirect(credential);
+      } else {
+        //the stitch client marks this user as logged in, add the user
+        //profile to state so we can view their name on the page
+        var userState = [];
+        userState[0] = stitchClient.auth.user.profile;
```

Comment:
May as well hold the whole user object in state in case you want to access their `id`